--- ssl_gen_statem.erl.orig
+++ ssl_gen_statem.erl
@@ -596,7 +596,7 @@
 		    [{reply, From, {ok, SelectedProtocol}}]);
 connection({call, From},
            {close, {Pid, _Timeout}},
-           #state{connection_env = #connection_env{terminated = closed} = CEnv,
+           #state{connection_env = #connection_env{terminated = false} = CEnv,
                  protocol_specific = PS} = State) ->
     {next_state, downgrade, State#state{connection_env =
                                             CEnv#connection_env{terminated = true,
